dallinger/prolific.py

```diff
@@ -17,12 +18,15 @@ class ProlificServiceException(Exception):
 class ProlificService:
     """Wrapper for Prolific REST API"""
 
-    def __init__(self, api_token: str, api_version: str, referer_header: str):
+    def __init__(
+        self, api_token: str, api_version: str, referer_header: str, sandbox: bool
```

Comment:
If you do this, maybe you should add an assertion in the `__init__` that throws an error if `sandbox=True`.

Reply:
Can you just write some documentation for the argument, then, explaining that it's retained here to keep the API consistent but is ignored within the class because Prolific doesn't have such a thing as sandbox?

Reply:
Or, alternatively, replace

```py
self.sandbox = sandbox
```

with

```py
_ = sandbox
```

to make it explicit that the argument is ignored. I think if I were doing it I'd probably do the latter.

Reply:
I added the documentation, however it's not ignored by Dallinger. E.g. have a look at `ProlificService.published_study(…)`:

```
        if mode == "live":
            logger.info(f"Publishing experiment {study_id} on Prolific...")
            return self.publish_study(study_id)
        else:
            logger.info(
                f"Sandboxing experiment {study_id} in Prolific (saved as draft, not public)..."
            )
            return draft
```

Reply:
I think it's good to add extra documentation to make this explicit. But the way I see this is that Dallinger needs a major refactoring to make recruiters independent of mturk. Which will also make the Prolific API less awkward. However, in my view this is not part of this MR.

Reply:
I checked the code in Dallinger and MTurkService calls it something different, it saves it as `is_sandbox`. So to the extent that you're trying to maintain cross-compatibility (i.e. in the hypothetical case that some other part of Dallinger calls `service.is_sandbox`) then it's not going to work anyway. if you want to keep it, I'd rename it to `is_sandbox`.